x-pack/test/api_integration/apis/ml/saved_objects/sync.ts

```diff
@@ -116,25 +160,31 @@ export default ({ getService }: FtrProviderContext) => {
       datafeedConfig2.datafeed_id = `different_${datafeedConfig2.datafeed_id}`;
       await ml.api.createDatafeedES(datafeedConfig2);
 
-      const body2 = await runRequest(USER.ML_POWERUSER_ALL_SPACES, 200);
-
-      // previous datafeed should be removed on first sync
-      expect(body2).to.eql({
-        datafeedsAdded: {},
-        datafeedsRemoved: { [adJobId1]: { success: true } },
-        savedObjectsCreated: {},
-        savedObjectsDeleted: {},
-      });
+      // check to see is a sync is needed
+      const syncNeeded3 = await runSyncCheckRequest(
+        USER.ML_POWERUSER_ALL_SPACES,
+        'anomaly-detector',
+        200
+      );
+      expect(syncNeeded3.result).to.eql(true, 'sync should be needed');
 
-      const body3 = await runRequest(USER.ML_POWERUSER_ALL_SPACES, 200);
+      const body2 = await runSyncRequest(USER.ML_POWERUSER_ALL_SPACES, 200);
 
       // new datafeed will be added on second sync
```

Comment:
With the adjusted tests, this comment should also be adjusted. Maybe something like `previous datafeed should be removed and new datafeed should be added on sync`?